src/ArmTemplates/Commands/Executors/ExtractorExecutor.cs

```diff
@@ -164,252 +150,267 @@ public async Task GenerateSplitAPITemplates(FileNames fileNames)
                     string tempFileFolder = string.Concat(@apiFileFolder, $@"/{apiName}");
                     System.IO.Directory.CreateDirectory(tempFileFolder);
                     // generate templates for each API
-                    await this.GenerateTemplates(new ExtractorParameters(this.extractorConfig, tempFileFolder), apiName, null, fileNames, null);
+                    await this.GenerateTemplates(tempFileFolder, apiName, null, null);
 
                     Logger.LogInformation($@"Finish extracting API {apiName}");
                 }
             }
         }
 
+        /// <summary>
+        /// Generates master template for each API within this version set and an extra master template to link these apis
+        /// </summary>
+        public async Task GenerateAPIVersionSetTemplates()
+        {
+            // get api dictionary and check api version set
+            var apiDictionary = await this.GetAllAPIsDictionary(this.extractorParameters.SourceApimName, this.extractorParameters.ResourceGroup);
+            if (!apiDictionary.ContainsKey(this.extractorParameters.ApiVersionSetName))
+            {
+                throw new NoApiVersionSetWithSuchNameFoundException("API Version Set with this name doesn't exist");
+            }
+            else
+            {
+                Logger.LogInformation("Start extracting the API version set {0}", this.extractorParameters.ApiVersionSetName);
+
+                foreach (string apiName in apiDictionary[this.extractorParameters.ApiVersionSetName])
+                {
+                    // generate seperate folder for each API
+                    string apiFileFolder = string.Concat(this.extractorParameters.FilesGenerationRootDirectory, $@"/{apiName}");
+                    System.IO.Directory.CreateDirectory(apiFileFolder);
+                    await this.GenerateTemplates(apiFileFolder, apiName, null, null);
+                }
+
+                // create master templates for this apiVersionSet 
+                string versionSetFolder = string.Concat(this.extractorParameters.FilesGenerationRootDirectory, this.extractorParameters.FileNames.VersionSetMasterFolder);
+                System.IO.Directory.CreateDirectory(versionSetFolder);
+                await this.GenerateTemplates(versionSetFolder, null, apiDictionary[this.extractorParameters.ApiVersionSetName], null);
+
+                Logger.LogInformation($@"Finish extracting APIVersionSet {this.extractorParameters.ApiVersionSetName}");
+            }
+        }
+
+        /// <summary>
+        /// Generates templates for multiple specified APIs
+        /// </summary>
+        public async Task GenerateMultipleAPIsTemplates()
```

Comment:
Many of these methods can be private, right?

Reply:
Ye, made most of them private. For specific extractors methods will still be exposed so that Executor methods for specific reasons can be called from outside.